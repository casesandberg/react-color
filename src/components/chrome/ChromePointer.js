'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import shallowCompare from 'react-addons-shallow-compare'

export class ChromePointer extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    return {
      'default': {
        picker: {
          width: '12px',
          height: '12px',
          borderRadius: '6px',
          transform: 'translate(-6px, -1px)',
          backgroundColor: 'rgb(248, 248, 248)',
          boxShadow: '0px 1px 4px 0px rgba(0, 0, 0, 0.37)',
        },
      },
    }
  }

  render(): any {
    return (
      <div is="picker"></div>
    )
  }

}

export default ChromePointer
