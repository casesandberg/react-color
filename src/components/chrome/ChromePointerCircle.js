'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import shallowCompare from 'react-addons-shallow-compare'

export class ChromePointerCircle extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    return {
      'default': {
        picker: {
          width: '12px',
          height: '12px',
          borderRadius: '6px',
          boxShadow: 'inset 0px 0px 0px 1px #fff',
          transform: 'translate(-6px, -6px)',
        },
      },
    }
  }

  render(): any {
    return (
      <div is="picker"></div>
    )
  }

}

export default ChromePointerCircle
