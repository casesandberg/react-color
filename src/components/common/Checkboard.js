'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import shallowCompare from 'react-addons-shallow-compare'

let _checkboardCache = {}

function renderCheckboard(c1: string, c2: string, size: number): any {
  if (typeof document == 'undefined') return null // Dont Render On Server
  var canvas: any = document.createElement('canvas')
  canvas.width = canvas.height = size * 2
  var ctx = canvas.getContext('2d')
  if (!ctx) return null // If no context can be found, return early.
  ctx.fillStyle = c1
  ctx.fillRect(0, 0, canvas.width, canvas.height)
  ctx.fillStyle = c2
  ctx.fillRect(0, 0, size, size)
  ctx.translate(size, size)
  ctx.fillRect(0, 0, size, size)
  return canvas.toDataURL()
}

function getCheckboard(c1: string, c2: string, size: number): any {
  var key = c1 + ',' + c2 + ',' + size

  if (_checkboardCache[key]) {
    return _checkboardCache[key]
  } else {
    var checkboard = renderCheckboard(c1, c2, size)
    _checkboardCache[key] = checkboard
    return checkboard
  }
}

export class Checkboard extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    var background = getCheckboard(this.props.white, this.props.grey, this.props.size)
    return {
      'default': {
        grid: {
          Absolute: '0px 0px 0px 0px',
          background: 'url(' + background + ') center left',
        },
      },
    }
  }

  render(): any {
    return (
      <div is="grid" ref="grid"></div>
    )
  }

}

Checkboard.defaultProps = {
  size: 8,
  white: '#fff',
  grey: '#e6e6e6',
}

export default Checkboard
