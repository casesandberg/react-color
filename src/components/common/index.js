export { default as Alpha } from './Alpha'
export { default as Checkboard } from './Checkboard'
export { default as EditableInput } from './EditableInput'
export { default as Hue } from './Hue'
export { default as Raised } from './Raised'
export { default as Saturation } from './Saturation'
export { default as ColorWrap } from './ColorWrap'
export { default as Swatch } from './Swatch'
