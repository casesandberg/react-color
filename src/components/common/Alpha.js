'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import shallowCompare from 'react-addons-shallow-compare'

import Checkboard from './Checkboard'

export class Alpha extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    return {
      'default': {
        alpha: {
          Absolute: '0px 0px 0px 0px',
          borderRadius: this.props.radius,
        },
        checkboard: {
          Absolute: '0px 0px 0px 0px',
          overflow: 'hidden',
        },
        gradient: {
          Absolute: '0px 0px 0px 0px',
          background: 'linear-gradient(to right, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 0) 0%, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 1) 100%)',
          boxShadow: this.props.shadow,
          borderRadius: this.props.radius,
        },
        container: {
          position: 'relative',
          zIndex: '2',
          height: '100%',
          margin: '0px 3px',
        },
        pointer: {
          zIndex: '2',
          position: 'absolute',
          left: this.props.rgb.a * 100 + '%',
        },
        slider: {
          width: '4px',
          borderRadius: '1px',
          height: '8px',
          boxShadow: '0px 0px 2px rgba(0, 0, 0, .6)',
          background: '#fff',
          marginTop: '1px',
          transform: 'translateX(-2px)',
        },
      },
    }
  }

  componentWillUnmount() {
    this.unbindEventListeners()
  }

  handleChange = (e: any, skip: boolean) => {
    !skip && e.preventDefault()
    var container = this.refs.container
    var containerWidth = container.clientWidth
    var left = (e.pageX || e.touches[0].pageX) - (container.getBoundingClientRect().left + window.pageXOffset)

    var a
    if (left < 0) {
      a = 0
    } else if (left > containerWidth) {
      a = 1
    } else {
      a = Math.round(left * 100 / containerWidth) / 100
    }

    if (this.props.a !== a) {
      this.props.onChange({
        h: this.props.hsl.h,
        s: this.props.hsl.s,
        l: this.props.hsl.l,
        a: a,
        source: 'rgb',
      })
    }
  }

  handleMouseDown = (e: any) => {
    this.handleChange(e, true)
    window.addEventListener('mousemove', this.handleChange)
    window.addEventListener('mouseup', this.handleMouseUp)
  }

  handleMouseUp = () => {
    this.unbindEventListeners()
  }

  unbindEventListeners = () => {
    window.removeEventListener('mousemove', this.handleChange)
    window.removeEventListener('mouseup', this.handleMouseUp)
  }

  render(): any {
    var pointer = <div is="slider" />

    if (this.props.pointer) {
      pointer = <this.props.pointer {...this.props} />
    }

    return (
      <div is="alpha">
        <div is="checkboard">
          <Checkboard />
        </div>
        <div is="gradient" />
        <div is="container" ref="container" onMouseDown={ this.handleMouseDown }
            onTouchMove={ this.handleChange }
            onTouchStart={ this.handleChange }>
          <div is="pointer" ref="pointer">
            { pointer }
          </div>
        </div>
      </div>
    )
  }
}

export default Alpha
