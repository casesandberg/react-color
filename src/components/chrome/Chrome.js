'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'

import { ColorWrap, Saturation, Hue, Alpha, Checkboard } from '../common'
import ChromeFields from './ChromeFields'
import ChromePointer from './ChromePointer'
import ChromePointerCircle from './ChromePointerCircle'
import shallowCompare from 'react-addons-shallow-compare'

export class Chrome extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    return {
      'default': {
        picker: {
          background: '#fff',
          borderRadius: '2px',
          boxShadow: '0 0 2px rgba(0,0,0,.3), 0 4px 8px rgba(0,0,0,.3)',
          boxSizing: 'initial',
          width: '225px',
          fontFamily: 'Menlo',
        },
        saturation: {
          width: '100%',
          paddingBottom: '55%',
          position: 'relative',
          borderRadius: '2px 2px 0 0',
          overflow: 'hidden',
        },
        Saturation: {
          radius: '2px 2px 0 0',
        },
        body: {
          padding: '16px 16px 12px',
        },
        controls: {
          display: 'flex',
        },
        color: {
          width: '32px',
        },
        swatch: {
          marginTop: '6px',
          width: '16px',
          height: '16px',
          borderRadius: '8px',
          position: 'relative',
          overflow: 'hidden',
        },
        active: {
          Absolute: '0px 0px 0px 0px',
          borderRadius: '8px',
          boxShadow: 'inset 0 0 0 1px rgba(0,0,0,.1)',
          background: 'rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', ' + this.props.rgb.a + ')',
          zIndex: '2',
        },
        toggles: {
          flex: '1',
        },
        hue: {
          height: '10px',
          position: 'relative',
          marginBottom: '8px',
        },
        Hue: {
          radius: '2px',
        },
        alpha: {
          height: '10px',
          position: 'relative',
        },
        Alpha: {
          radius: '2px',
        },
      },
      'disableAlpha': {
        color: {
          width: '22px',
        },
        alpha: {
          display: 'none',
        },
        hue: {
          marginBottom: '0px',
        },
        swatch: {
          width: '10px',
          height: '10px',
          marginTop: '0px',
        },
      },
    }
  }

  handleChange = (data: any) => {
    this.props.onChange(data)
  }

  render(): any {
    return (
      <div is="picker">
        <div is="saturation">
          <Saturation is="Saturation" {...this.props} pointer={ ChromePointerCircle } onChange={ this.handleChange }/>
        </div>
        <div is="body">
          <div is="controls" className="flexbox-fix">
            <div is="color">
              <div is="swatch">
                <div is="active" />
                <Checkboard />
              </div>
            </div>
            <div is="toggles">
              <div is="hue">
                <Hue is="Hue" {...this.props} pointer={ ChromePointer } onChange={ this.handleChange } />
              </div>
              <div is="alpha">
                <Alpha is="Alpha" {...this.props} pointer={ ChromePointer } onChange={ this.handleChange } />
              </div>
            </div>
          </div>
          <ChromeFields {...this.props} onChange={ this.handleChange } disableAlpha={ this.props.disableAlpha } />
        </div>
      </div>
    )
  }

}

export default ColorWrap(Chrome)
