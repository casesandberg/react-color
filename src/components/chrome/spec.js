/* global test, jest, expect */

import React from 'react'
import renderer from 'react-test-renderer'
import color, { red } from '../../helpers/color'
import { mount } from 'enzyme'

import Chrome from './Chrome'
import ChromeFields from './ChromeFields'
import ChromePointer from './ChromePointer'
import ChromePointerCircle from './ChromePointerCircle'
import { Alpha } from '../common'


test('Chrome renders correctly', () => {
  const tree = renderer.create(
    <Chrome { ...red } />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Chrome onChange events correctly', () => {
  const changeSpy = jest.fn((data) => {
    expect(color.simpleCheckForValidColor(data)).toBeTruthy()
  })
  const tree = mount(
    <Chrome { ...red } onChange={ changeSpy } />,
  )
  expect(changeSpy).toHaveBeenCalledTimes(0)

  // check the Alpha component
  const alphaCommon = tree.find(Alpha)
  alphaCommon.at(0).childAt(2).simulate('mouseDown', {
    pageX: 100,
    pageY: 10,
  })
  expect(changeSpy).toHaveBeenCalledTimes(1)

  // TODO: check the Hue component
  // TODO: check the ChromeFields
  // TODO: check Saturation
})

// test('Chrome renders on server correctly', () => {
//   const tree = renderer.create(
//     <Chrome renderers={{ canvas }} { ...red } />
//   ).toJSON()
//   expect(tree).toMatchSnapshot()
// })

test('ChromeFields renders correctly', () => {
  const tree = renderer.create(
    <ChromeFields { ...red } />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('ChromePointer renders correctly', () => {
  const tree = renderer.create(
    <ChromePointer />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('ChromePointerCircle renders correctly', () => {
  const tree = renderer.create(
    <ChromePointerCircle />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Chrome renders custom styles correctly', () => {
  const tree = renderer.create(
    <Chrome styles={{ default: { picker: { boxShadow: 'none' } } }} />,
  ).toJSON()
  expect(tree.props.style.boxShadow).toBe('none')
})

test('Chrome renders correctly with width', () => {
  const tree = renderer.create(
    <Chrome width={300} />,
  ).toJSON()
  expect(tree.props.style.width).toBe(300)
});