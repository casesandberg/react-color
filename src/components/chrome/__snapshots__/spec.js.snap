// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chrome renders correctly 1`] = `
<div
  className="chrome-picker "
  style={
    Object {
      "MozBorderRadius": "2px",
      "MozBoxShadow": "0 0 2px rgba(0,0,0,.3), 0 4px 8px rgba(0,0,0,.3)",
      "OBorderRadius": "2px",
      "OBoxShadow": "0 0 2px rgba(0,0,0,.3), 0 4px 8px rgba(0,0,0,.3)",
      "WebkitBorderRadius": "2px",
      "WebkitBoxShadow": "0 0 2px rgba(0,0,0,.3), 0 4px 8px rgba(0,0,0,.3)",
      "background": "#fff",
      "borderRadius": "2px",
      "boxShadow": "0 0 2px rgba(0,0,0,.3), 0 4px 8px rgba(0,0,0,.3)",
      "boxSizing": "initial",
      "fontFamily": "Menlo",
      "msBorderRadius": "2px",
      "msBoxShadow": "0 0 2px rgba(0,0,0,.3), 0 4px 8px rgba(0,0,0,.3)",
      "width": 225,
    }
  }
>
  <div
    style={
      Object {
        "MozBorderRadius": "2px 2px 0 0",
        "OBorderRadius": "2px 2px 0 0",
        "WebkitBorderRadius": "2px 2px 0 0",
        "borderRadius": "2px 2px 0 0",
        "msBorderRadius": "2px 2px 0 0",
        "overflow": "hidden",
        "paddingBottom": "55%",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      onMouseDown={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "MozBorderRadius": undefined,
          "OBorderRadius": undefined,
          "WebkitBorderRadius": undefined,
          "background": "hsl(249.99999999999994,100%, 50%)",
          "borderRadius": undefined,
          "bottom": "0px",
          "left": "0px",
          "msBorderRadius": undefined,
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    >
      <style>
        
                  .saturation-white {
                    background: -webkit-linear-gradient(to right, #fff, rgba(255,255,255,0));
                    background: linear-gradient(to right, #fff, rgba(255,255,255,0));
                  }
                  .saturation-black {
                    background: -webkit-linear-gradient(to top, #000, rgba(0,0,0,0));
                    background: linear-gradient(to top, #000, rgba(0,0,0,0));
                  }
                
      </style>
      <div
        className="saturation-white"
        style={
          Object {
            "MozBorderRadius": undefined,
            "OBorderRadius": undefined,
            "WebkitBorderRadius": undefined,
            "borderRadius": undefined,
            "bottom": "0px",
            "left": "0px",
            "msBorderRadius": undefined,
            "position": "absolute",
            "right": "0px",
            "top": "0px",
          }
        }
      >
        <div
          className="saturation-black"
          style={
            Object {
              "MozBorderRadius": undefined,
              "MozBoxShadow": undefined,
              "OBorderRadius": undefined,
              "OBoxShadow": undefined,
              "WebkitBorderRadius": undefined,
              "WebkitBoxShadow": undefined,
              "borderRadius": undefined,
              "bottom": "0px",
              "boxShadow": undefined,
              "left": "0px",
              "msBorderRadius": undefined,
              "msBoxShadow": undefined,
              "position": "absolute",
              "right": "0px",
              "top": "0px",
            }
          }
        />
        <div
          style={
            Object {
              "cursor": "default",
              "left": "66.66666666666667%",
              "position": "absolute",
              "top": "70%",
            }
          }
        >
          <div
            style={
              Object {
                "MozBorderRadius": "6px",
                "MozBoxShadow": "inset 0 0 0 1px #fff",
                "MozTransform": "translate(-6px, -6px)",
                "OBorderRadius": "6px",
                "OBoxShadow": "inset 0 0 0 1px #fff",
                "OTransform": "translate(-6px, -6px)",
                "WebkitBorderRadius": "6px",
                "WebkitBoxShadow": "inset 0 0 0 1px #fff",
                "WebkitTransform": "translate(-6px, -6px)",
                "borderRadius": "6px",
                "boxShadow": "inset 0 0 0 1px #fff",
                "height": "12px",
                "msBorderRadius": "6px",
                "msBoxShadow": "inset 0 0 0 1px #fff",
                "msTransform": "translate(-6px, -6px)",
                "transform": "translate(-6px, -6px)",
                "width": "12px",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px 16px 12px",
      }
    }
  >
    <div
      className="flexbox-fix"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        style={
          Object {
            "width": "32px",
          }
        }
      >
        <div
          style={
            Object {
              "MozBorderRadius": "8px",
              "OBorderRadius": "8px",
              "WebkitBorderRadius": "8px",
              "borderRadius": "8px",
              "height": "16px",
              "marginTop": "6px",
              "msBorderRadius": "8px",
              "overflow": "hidden",
              "position": "relative",
              "width": "16px",
            }
          }
        >
          <div
            style={
              Object {
                "MozBorderRadius": "8px",
                "MozBoxShadow": "inset 0 0 0 1px rgba(0,0,0,.1)",
                "OBorderRadius": "8px",
                "OBoxShadow": "inset 0 0 0 1px rgba(0,0,0,.1)",
                "WebkitBorderRadius": "8px",
                "WebkitBoxShadow": "inset 0 0 0 1px rgba(0,0,0,.1)",
                "background": "rgba(34, 25, 77, 1)",
                "borderRadius": "8px",
                "bottom": "0px",
                "boxShadow": "inset 0 0 0 1px rgba(0,0,0,.1)",
                "left": "0px",
                "msBorderRadius": "8px",
                "msBoxShadow": "inset 0 0 0 1px rgba(0,0,0,.1)",
                "position": "absolute",
                "right": "0px",
                "top": "0px",
                "zIndex": "2",
              }
            }
          />
          <div
            style={
              Object {
                "MozBorderRadius": undefined,
                "MozBoxShadow": undefined,
                "OBorderRadius": undefined,
                "OBoxShadow": undefined,
                "WebkitBorderRadius": undefined,
                "WebkitBoxShadow": undefined,
                "background": "url(null) center left",
                "borderRadius": undefined,
                "bottom": "0px",
                "boxShadow": undefined,
                "left": "0px",
                "msBorderRadius": undefined,
                "msBoxShadow": undefined,
                "position": "absolute",
                "right": "0px",
                "top": "0px",
              }
            }
          />
        </div>
      </div>
      <div
        style={
          Object {
            "MozBoxFlex": "1",
            "WebkitBoxFlex": "1",
            "WebkitFlex": "1",
            "flex": "1",
            "msFlex": "1",
          }
        }
      >
        <div
          style={
            Object {
              "height": "10px",
              "marginBottom": "8px",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "MozBorderRadius": undefined,
                "MozBoxShadow": undefined,
                "OBorderRadius": undefined,
                "OBoxShadow": undefined,
                "WebkitBorderRadius": undefined,
                "WebkitBoxShadow": undefined,
                "borderRadius": undefined,
                "bottom": "0px",
                "boxShadow": undefined,
                "left": "0px",
                "msBorderRadius": undefined,
                "msBoxShadow": undefined,
                "position": "absolute",
                "right": "0px",
                "top": "0px",
              }
            }
          >
            <div
              className="hue-horizontal"
              onMouseDown={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "MozBorderRadius": undefined,
                  "OBorderRadius": undefined,
                  "WebkitBorderRadius": undefined,
                  "borderRadius": undefined,
                  "height": "100%",
                  "msBorderRadius": undefined,
                  "padding": "0 2px",
                  "position": "relative",
                }
              }
            >
              <style>
                
                            .hue-horizontal {
                              background: linear-gradient(to right, #f00 0%, #ff0 17%, #0f0
                                33%, #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);
                              background: -webkit-linear-gradient(to right, #f00 0%, #ff0
                                17%, #0f0 33%, #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);
                            }
                
                            .hue-vertical {
                              background: linear-gradient(to top, #f00 0%, #ff0 17%, #0f0 33%,
                                #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);
                              background: -webkit-linear-gradient(to top, #f00 0%, #ff0 17%,
                                #0f0 33%, #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);
                            }
                          
              </style>
              <div
                style={
                  Object {
                    "left": "69.44444444444443%",
                    "position": "absolute",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "MozBorderRadius": "6px",
                      "MozBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
                      "MozTransform": "translate(-6px, -1px)",
                      "OBorderRadius": "6px",
                      "OBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
                      "OTransform": "translate(-6px, -1px)",
                      "WebkitBorderRadius": "6px",
                      "WebkitBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
                      "WebkitTransform": "translate(-6px, -1px)",
                      "backgroundColor": "rgb(248, 248, 248)",
                      "borderRadius": "6px",
                      "boxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
                      "height": "12px",
                      "msBorderRadius": "6px",
                      "msBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
                      "msTransform": "translate(-6px, -1px)",
                      "transform": "translate(-6px, -1px)",
                      "width": "12px",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "height": "10px",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "MozBorderRadius": undefined,
                "OBorderRadius": undefined,
                "WebkitBorderRadius": undefined,
                "borderRadius": undefined,
                "bottom": "0px",
                "left": "0px",
                "msBorderRadius": undefined,
                "position": "absolute",
                "right": "0px",
                "top": "0px",
              }
            }
          >
            <div
              style={
                Object {
                  "MozBorderRadius": undefined,
                  "OBorderRadius": undefined,
                  "WebkitBorderRadius": undefined,
                  "borderRadius": undefined,
                  "bottom": "0px",
                  "left": "0px",
                  "msBorderRadius": undefined,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": "0px",
                  "top": "0px",
                }
              }
            >
              <div
                style={
                  Object {
                    "MozBorderRadius": undefined,
                    "MozBoxShadow": undefined,
                    "OBorderRadius": undefined,
                    "OBoxShadow": undefined,
                    "WebkitBorderRadius": undefined,
                    "WebkitBoxShadow": undefined,
                    "background": "url(null) center left",
                    "borderRadius": undefined,
                    "bottom": "0px",
                    "boxShadow": undefined,
                    "left": "0px",
                    "msBorderRadius": undefined,
                    "msBoxShadow": undefined,
                    "position": "absolute",
                    "right": "0px",
                    "top": "0px",
                  }
                }
              />
            </div>
            <div
              style={
                Object {
                  "MozBorderRadius": undefined,
                  "MozBoxShadow": undefined,
                  "OBorderRadius": undefined,
                  "OBoxShadow": undefined,
                  "WebkitBorderRadius": undefined,
                  "WebkitBoxShadow": undefined,
                  "background": "linear-gradient(to right, rgba(34,25,77, 0) 0%,
                           rgba(34,25,77, 1) 100%)",
                  "borderRadius": undefined,
                  "bottom": "0px",
                  "boxShadow": undefined,
                  "left": "0px",
                  "msBorderRadius": undefined,
                  "msBoxShadow": undefined,
                  "position": "absolute",
                  "right": "0px",
                  "top": "0px",
                }
              }
            />
            <div
              onMouseDown={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "height": "100%",
                  "margin": "0 3px",
                  "position": "relative",
                }
              }
            >
              <div
                style={
                  Object {
                    "left": "100%",
                    "position": "absolute",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "MozBorderRadius": "6px",
                      "MozBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
                      "MozTransform": "translate(-6px, -1px)",
                      "OBorderRadius": "6px",
                      "OBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
                      "OTransform": "translate(-6px, -1px)",
                      "WebkitBorderRadius": "6px",
                      "WebkitBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
                      "WebkitTransform": "translate(-6px, -1px)",
                      "backgroundColor": "rgb(248, 248, 248)",
                      "borderRadius": "6px",
                      "boxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
                      "height": "12px",
                      "msBorderRadius": "6px",
                      "msBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
                      "msTransform": "translate(-6px, -1px)",
                      "transform": "translate(-6px, -1px)",
                      "width": "12px",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="flexbox-fix"
      style={
        Object {
          "display": "flex",
          "paddingTop": "16px",
        }
      }
    >
      <div
        className="flexbox-fix"
        style={
          Object {
            "MozBoxFlex": "1",
            "WebkitBoxFlex": "1",
            "WebkitFlex": "1",
            "display": "flex",
            "flex": "1",
            "marginLeft": "-6px",
            "msFlex": "1",
          }
        }
      >
        <div
          style={
            Object {
              "paddingLeft": "6px",
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <input
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder={undefined}
              spellCheck="false"
              style={
                Object {
                  "MozBorderRadius": "2px",
                  "MozBoxShadow": "inset 0 0 0 1px #dadada",
                  "OBorderRadius": "2px",
                  "OBoxShadow": "inset 0 0 0 1px #dadada",
                  "WebkitBorderRadius": "2px",
                  "WebkitBoxShadow": "inset 0 0 0 1px #dadada",
                  "border": "none",
                  "borderRadius": "2px",
                  "boxShadow": "inset 0 0 0 1px #dadada",
                  "color": "#333",
                  "fontSize": "11px",
                  "height": "21px",
                  "msBorderRadius": "2px",
                  "msBoxShadow": "inset 0 0 0 1px #dadada",
                  "textAlign": "center",
                  "width": "100%",
                }
              }
              value="#22194D"
            />
            <span
              onMouseDown={[Function]}
              style={
                Object {
                  "color": "#969696",
                  "display": "block",
                  "fontSize": "11px",
                  "lineHeight": "11px",
                  "marginTop": "12px",
                  "textAlign": "center",
                  "textTransform": "uppercase",
                }
              }
            >
              hex
            </span>
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "position": "relative",
            "textAlign": "right",
            "width": "32px",
          }
        }
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "marginRight": "-4px",
              "marginTop": "12px",
              "position": "relative",
            }
          }
        >
          <svg
            onMouseEnter={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "MozBorderRadius": "5px",
                "OBorderRadius": "5px",
                "WebkitBorderRadius": "5px",
                "border": "1px transparent solid",
                "borderRadius": "5px",
                "fill": "#333",
                "height": "24px",
                "msBorderRadius": "5px",
                "width": "24px",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12,18.17L8.83,15L7.42,16.41L12,21L16.59,16.41L15.17,15M12,5.83L15.17,9L16.58,7.59L12,3L7.41,7.59L8.83,9L12,5.83Z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ChromeFields renders correctly 1`] = `
<div
  className="flexbox-fix"
  style={
    Object {
      "display": "flex",
      "paddingTop": "16px",
    }
  }
>
  <div
    className="flexbox-fix"
    style={
      Object {
        "MozBoxFlex": "1",
        "WebkitBoxFlex": "1",
        "WebkitFlex": "1",
        "display": "flex",
        "flex": "1",
        "marginLeft": "-6px",
        "msFlex": "1",
      }
    }
  >
    <div
      style={
        Object {
          "paddingLeft": "6px",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder={undefined}
          spellCheck="false"
          style={
            Object {
              "MozBorderRadius": "2px",
              "MozBoxShadow": "inset 0 0 0 1px #dadada",
              "OBorderRadius": "2px",
              "OBoxShadow": "inset 0 0 0 1px #dadada",
              "WebkitBorderRadius": "2px",
              "WebkitBoxShadow": "inset 0 0 0 1px #dadada",
              "border": "none",
              "borderRadius": "2px",
              "boxShadow": "inset 0 0 0 1px #dadada",
              "color": "#333",
              "fontSize": "11px",
              "height": "21px",
              "msBorderRadius": "2px",
              "msBoxShadow": "inset 0 0 0 1px #dadada",
              "textAlign": "center",
              "width": "100%",
            }
          }
          value="#FF0000"
        />
        <span
          onMouseDown={[Function]}
          style={
            Object {
              "color": "#969696",
              "display": "block",
              "fontSize": "11px",
              "lineHeight": "11px",
              "marginTop": "12px",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
        >
          hex
        </span>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "position": "relative",
        "textAlign": "right",
        "width": "32px",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "marginRight": "-4px",
          "marginTop": "12px",
          "position": "relative",
        }
      }
    >
      <svg
        onMouseEnter={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "MozBorderRadius": "5px",
            "OBorderRadius": "5px",
            "WebkitBorderRadius": "5px",
            "border": "1px transparent solid",
            "borderRadius": "5px",
            "fill": "#333",
            "height": "24px",
            "msBorderRadius": "5px",
            "width": "24px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12,18.17L8.83,15L7.42,16.41L12,21L16.59,16.41L15.17,15M12,5.83L15.17,9L16.58,7.59L12,3L7.41,7.59L8.83,9L12,5.83Z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`ChromePointer renders correctly 1`] = `
<div
  style={
    Object {
      "MozBorderRadius": "6px",
      "MozBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "MozTransform": "translate(-6px, -1px)",
      "OBorderRadius": "6px",
      "OBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "OTransform": "translate(-6px, -1px)",
      "WebkitBorderRadius": "6px",
      "WebkitBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "WebkitTransform": "translate(-6px, -1px)",
      "backgroundColor": "rgb(248, 248, 248)",
      "borderRadius": "6px",
      "boxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "height": "12px",
      "msBorderRadius": "6px",
      "msBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "msTransform": "translate(-6px, -1px)",
      "transform": "translate(-6px, -1px)",
      "width": "12px",
    }
  }
/>
`;

exports[`ChromePointerCircle renders correctly 1`] = `
<div
  style={
    Object {
      "MozBorderRadius": "6px",
      "MozBoxShadow": "inset 0 0 0 1px #fff",
      "MozTransform": "translate(-6px, -6px)",
      "OBorderRadius": "6px",
      "OBoxShadow": "inset 0 0 0 1px #fff",
      "OTransform": "translate(-6px, -6px)",
      "WebkitBorderRadius": "6px",
      "WebkitBoxShadow": "inset 0 0 0 1px #fff",
      "WebkitTransform": "translate(-6px, -6px)",
      "borderRadius": "6px",
      "boxShadow": "inset 0 0 0 1px #fff",
      "height": "12px",
      "msBorderRadius": "6px",
      "msBoxShadow": "inset 0 0 0 1px #fff",
      "msTransform": "translate(-6px, -6px)",
      "transform": "translate(-6px, -6px)",
      "width": "12px",
    }
  }
/>
`;
